unit/optional.cpp

```diff
@@ -0,0 +1,37 @@
+/*******************************************************************\
+
+ Module: nonstd::optional unit tests
+
+ Author: Diffblue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <catch.hpp>
+#include <util/optional.h>
+
+TEST_CASE("Optional without a value")
```

Comment:
As per the [guidelines](https://github.com/diffblue/cbmc/blob/master/CODING_STANDARD.md#unit-tests) could you tag your tests appropriately:

```
[core][util][optional]
```